modules/distribution/resources/migration-1.8.0_to_1.9.0/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/MigrateFrom18to19.java

```diff
@@ -0,0 +1,702 @@
+                    responseObject = new JSONObject();
+                    JSONArray responseMessages = (JSONArray) operationObject.get("responseMessages");
+                    for (Object responseMessage : responseMessages) {
+                        JSONObject errorObj = (JSONObject) responseMessage;
+                        responseObject.put(errorObj.get("code"), errorObj.get("message"));
+                    }
+                }
+                if (responseObject == null) {
+                    //set a default response message since this is required field
+                    responseObject = (JSONObject) jsonParser.parse(Constants.DEFAULT_RESPONSE);
+                }
+                swagger2OperationsObj.put("responses", responseObject);
+            }
+            pathsObj.put(key, pathItemObj);
+        }
+        return pathsObj;
+    }
+
+
+    /**
+     * This method is used to migrate rxt
+     * This adds three new attributes to the api rxt
+     *
+     * @throws APIMigrationException
+     */
+    void rxtMigration() throws APIMigrationException {
+        log.info("Rxt migration for API Manager 1.9.0 started.");
+        try {
+            for (Tenant tenant : tenantsArray) {
+                String adminName = ServiceHolder.getRealmService().getTenantUserRealm(tenant.getId())
+                        .getRealmConfiguration().getAdminUserName();
+                ServiceHolder.getTenantRegLoader().loadTenantRegistry(tenant.getId());
+                Registry registry = ServiceHolder.getRegistryService().getGovernanceUserRegistry(adminName, tenant
+                        .getId());
+                GenericArtifactManager artifactManager = APIUtil.getArtifactManager(registry, APIConstants.API_KEY);
+                GenericArtifactManager manager = new GenericArtifactManager(registry, "api");
+                GovernanceUtils.loadGovernanceArtifacts((UserRegistry) registry);
+                GenericArtifact[] artifacts = manager.getAllGenericArtifacts();
+                for (GenericArtifact artifact : artifacts) {
+                    API api = APIUtil.getAPI(artifact, registry);
+                    artifact.addAttribute("overview_contextTemplate", api.getContext() + "/{version}");
+                    artifact.addAttribute("overview_environments", "");
+                    artifact.addAttribute("overview_versionType", "");
+
+                    artifactManager.updateGenericArtifact(artifact);
+                }
+            }
+        } catch (APIManagementException e) {
+            ResourceUtil.handleException("Error occurred while reading API from the artifact ", e);
+        } catch (RegistryException e) {
+            ResourceUtil.handleException("Error occurred while accessing the registry", e);
+        } catch (UserStoreException e) {
+            ResourceUtil.handleException("Error occurred while reading tenant information", e);
+        }
+        if (log.isDebugEnabled()) {
```

Comment:
This should be an info log since the start of the method as an info log as well.